codegen/gen/gen.go

```diff
@@ -0,0 +1,141 @@
+// Generator that converts a schema yaml file to schema and type definitions golang files.
+// Usage of gen:
+// 	gen schema_input.yaml
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"go/format"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	"gopkg.in/yaml.v3"
+)
+
+const typeDefImports = `package arcaflow_plugin_service
+
+import (
+    v1 "k8s.io/api/core/v1"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+`
+
+type property struct {
+	Type struct {
+		TypeID string `yaml:"type_id"`
+		Id     string `yaml:"id"`
+	} `yaml:"type"`
+	Display struct {
+		Name        string `yaml:"name"`
+		Description string `yaml:"description"`
+	} `yaml:"display"`
+	Required bool `yaml:"required"`
+}
+
+type schema struct {
+	Steps struct {
+		Create struct {
+			Id    string `yaml:"id"`
+			Input struct {
+				Objects map[string]struct {
+					Id         string               `yaml:"id"`
+					Properties map[string]*property `yaml:"properties"`
+				} `yaml:"objects"`
+			} `yaml:"input"`
+		} `yaml:"create"`
+	} `yaml:"steps"`
+}
+
+func mustGenerateTypeDef(schema schema) []byte {
+	var (
+		bft = new(bytes.Buffer) // Accumulated output for type def
+		bwt = bufio.NewWriter(bft)
+	)
+	fmt.Fprintf(bwt, "// Code generated by \"gen %s\"\n", strings.Join(os.Args[1:], " "))
+	fmt.Fprint(bwt, typeDefImports)
+	for o, ov := range schema.Steps.Create.Input.Objects {
+		if o == "ObjectMeta" { // Object "ObjectMeta" should be ignored
```

Comment:
I guess you are refering #L63: `if o == "ObjectMeta"`, correct?
If that is the case, I was assuming from our slack conversation that I could always assume:
 - to include this as an import: `metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"`
 - to ignore the `ObjectMeta` object from the input
 
If the second assumption is not correct, I can change a bit the logic for this function to avoid hardcoding `ObjectMeta`:
  - from the schema input file, extract all objects
  - if there is a metadata property in an object pointing to a second object, do not create a struct for the second object (that is the case for _ObjectMeta_)
  - create structs for the other objects